CONTRIBUTING.md

```diff
@@ -2,39 +2,75 @@
+- Feel free to open a Pull Request early so that a discussion can be had as
+  changes are developed.
 
 ## Commit Messages and Pull Request Titles
 
-We use the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification to standardize our commit history. To enforce this convention on commit messages and/or pull request titles, we use the [Semantic Pull Requests](https://github.com/probot/semantic-pull-requests) bot.
+We use the [Conventional Commits][] specification to standardize our commit
+history. To enforce this convention on commit messages and/or pull request
+titles, we use the [Semantic Pull Requests][] bot.
 
-The commit message summary (or pull request title) is constructed by prepending the type of change being made (e.g., feat, fix, refactor), followed by an imperative, present tense sentence (without a period).
-Example: `fix: make header bold`.
+The commit message summary (or pull request title) is constructed by prepending
+the type of change being made (e.g., feat, fix, refactor), followed by an
+imperative, present tense sentence (without a period). Example:
+`fix: make header bold`.
 
-If you are still working on your pull request, prepend `WIP:` to indicate that it's work in progress.
+If you are still working on your pull request, prepend `WIP:` to indicate that
+it's work in progress.
 
 ### Pull Request Title
 
-Same as commit messages, prepend the type of change being made (refactor, fix, chore, feat, etc.) 
-Example: `docs: add linux setup instructions`
+Same as commit messages, prepend the type of change being made (refactor, fix,
+chore, feat, etc.) Example: `docs: add linux setup instructions`
 
 ## Documentation
 
-Most of the documentation are automatically generated. We utilize `typedoc` that uses tags inline
-with classes in order to generate. The nice thing about this is that documentation will be visible
-when using Typescript in editors as well.
+Most of the documentation are automatically generated. We utilize `typedoc` that
+uses tags inline with classes in order to generate. The nice thing about this is
+that documentation will be visible when using Typescript in editors as well.
+
+You can generate documentation with the following command:
+
+```shell
+▶ yarn generate-docs
+```
 
 However, we love to have examples and guides as well.
 
+## Releasing
+
+This package utilizes [semantic release][] to manage releasing. We have three
+release channels, `main`, `next` and `develop`. Our Github Actions workflow
+automatically releases a canary release based on the content at `develop` as
+long as at least one commit message follows [Conventional Commits][].
+
+When the team decides too, `develop` is merged into `next` or `main`, dependant
+on the changes.
+
+- `develop` should be considered as unstable code (not ready for production).
+  e.g. `v1.0.3-develop.5`
+- `next` used for the next major version. e.g. `v2.0.0-next.1`
```

Comment:
I don't understand the difference of merging into `develop` and `next` and thus don't understand the need for `next`. Is it not possible to merge pull requests to `develop` and then merge `develop` into `main` whenever we want a stable release? Then all unstable releases come from `develop` and can be named `v2.0.0-next.1` or `v1.0.3-develop.5` depending on the existence of git tags?

Reply:
It is possible to merge into main whenever we want a stable release. However, in cases where we are moving over to the next major version, is when we would need `next`. 

So, let's say that we are running `v1` on `main` branch. And we want to do some backwards in-compatible changes, but iterate on that a while, while still releasing on `v1`. This case is perfect for `next`.

However, all that being said. This is not required right now. Maybe we should remove it _for now_ and add it if/when we have that case?

Reply:
> It is possible to merge into main whenever we want a stable release. However, in cases where we are moving over to the next major version, is when we would need `next`.
> 
> So, let's say that we are running `v1` on `main` branch. And we want to do some backwards in-compatible changes, but iterate on that a while, while still releasing on `v1`. This case is perfect for `next`.

Why can't `develop` be used for this purpose?

> However, all that being said. This is not required right now. Maybe we should remove it _for now_ and add it if/when we have that case?

Yes, I think we should manage without it. `main` for the current, stable version, `develop` for the next, unstable, possibly backwards incompatible, version.

Reply:
> Why can't develop be used for this purpose?
It can.

> Yes, I think we should manage without it. main for the current, stable version, develop for the next, unstable, possibly backwards incompatible, version.

Let me change my PR to reflect that!